scripts/polkadotjs_endpoints_to_preconfigured.py

```diff
@@ -0,0 +1,248 @@
+    if not array_matches:
+        raise ValueError("No array found in the input file")
+    json_objects = []
+    for i, array_content in enumerate(array_matches):
+        # Split the array content into individual objects
+        objects = find_objects(array_content)
+
+        for obj in objects:
+            # Remove comments
+            obj_lines = [line for line in obj.split('\n') if not line.strip().startswith('//')]
+            cleaned_obj = '\n'.join(obj_lines)
+
+            # Convert to valid JSON
+            cleaned_obj = re.sub(r"'", '"', cleaned_obj)  # Replace single quotes with double quotes
+            cleaned_obj = re.sub(r'(?<!")(\b\w+\b)(?=\s*:)', r'"\1"', cleaned_obj)  # Add quotes to keys
+            cleaned_obj = re.sub(r",\s*}", "}", cleaned_obj)  # Remove trailing commas
+            cleaned_obj = re.sub(r":\s*([a-zA-Z]\w*)", r': "\1"', cleaned_obj)  # Quote unquoted string values
+            cleaned_obj = re.sub(r"\n", '', cleaned_obj)
+
+            # Parse the cleaned object
+            try:
+                json_obj = json.loads(cleaned_obj)
+                json_objects.append(json_obj)
+            except json.JSONDecodeError as e:
+                print(f"Error parsing object: {e}")
+                print(f"Problematic object: {cleaned_obj}")
+
+    save_json_file(json_file_path, json_objects)
+    print(f"Conversion completed. Output written to {json_file_path}")
+
+    return json_objects
+
+
+def create_chain_data(chain_object):
+    providers = chain_object.get("providers", {})
+    # Get the first provider (if any)
+    # TODO: Iterate through all nodes available until connection is established
+    first_provider = None
+    if providers:
+        first_provider_value = next(iter(providers.values()), None)
+        wss_url = first_provider_value.strip("'")
+        try:
+            substrate = create_connection_by_url(wss_url)
+
+            json_property = get_properties(substrate)
+            providers = chain_object.get("providers")
+
+            chain_data = {
+                "chainId": json_property.chainId[2:],
+                "name": json_property.name,
+                "assets": [{
+                    "assetId": 0,
+                    "symbol": json_property.symbol,
+                    "precision": json_property.precision,
+                    "icon": "https://raw.githubusercontent.com/novasamatech/nova-utils/master/icons/tokens/white/Default.svg"
+                }],
+                "nodes": [{"url": url, "name": name} for name, url in providers.items()],
+                "addressPrefix": json_property.ss58Format
+            }
+            return chain_data
+        except Exception as err:
+            # If there's a failure, print a warning and skip the connection
+            print(f"⚠️ Can't connect by {wss_url}, check if it is available? \n {err}")
+            # Do not raise the exception; instead, return None or handle it accordingly
+            return None
```

Comment:
in order to improve readability, lets remove one nested level by return None if no providers, also first_provider does not use:
```suggestion
def create_chain_data(chain_object):
    providers = chain_object.get("providers", {})
    if not providers:
        return None

    # Get the first provider (if any)
    # TODO: Iterate through all nodes available until connection is established
    first_provider_value = next(iter(providers.values()), None)
    wss_url = first_provider_value.strip("'")
    try:
        substrate = create_connection_by_url(wss_url)

        json_property = get_properties(substrate)
        providers = chain_object.get("providers")

        chain_data = {
            "chainId": json_property.chainId[2:],
            "name": json_property.name,
            "assets": [{
                "assetId": 0,
                "symbol": json_property.symbol,
                "precision": json_property.precision,
                "icon": "https://raw.githubusercontent.com/novasamatech/nova-utils/master/icons/tokens/white/Default.svg"
            }],
            "nodes": [{"url": url, "name": name} for name, url in providers.items()],
            "addressPrefix": json_property.ss58Format
        }
        return chain_data
    except Exception as err:
        # If there's a failure, print a warning and skip the connection
        print(f"⚠️ Can't connect by {wss_url}, check if it is available? \n {err}")
        # Do not raise the exception; instead, return None or handle it accordingly
        return None
```